node/node.go

```diff
@@ -167,7 +168,7 @@ func (n *Node) headerPublishLoop(ctx context.Context) {
 				n.Logger.Error("failed to gossip block header", "error", err)
 			}
 		case <-ctx.Done():
-			break
+			cont = false
```

Comment:
Couldn't we just return here instead? There is no point in breaking out of the for loop or discontinuing otherwise bc there is no code after that loop. cc @tzdybal

Reply:
Are you worried about performance or readability? For performance, the compiler can optimize it. For readability, I personally prefer "breaking out of a for loop" to be consistent, regardless of the existence or lack thereof of post-conditions outside the loop. Thoughts?

Reply:
I think it is very readable to `return` as it's very explicit that this breaks out of the loop (as well as out of the function). With the cont = false you have to "scroll" up before the for again to realize that it is what is happening.

Reply:
Alright, reader knows best

Reply:
Fixed in 3750541